ci: CI fails when enabling clipboard on the actions

Caused by xvfb and clipboard on the github actions.

## Log
```
  xvfb-run --auto-servernum nimble test --verbose -Y
  shell: /usr/bin/bash -e {0}
  env:
    NIM_VERSION: stable
    PATH: /home/runner/.nimble/bin:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/home/runner/.local/bin:/opt/pipx_bin:/home/runner/.cargo/bin:/home/runner/.config/composer/vendor/bin:/usr/local/.ghcup/bin:/home/runner/.dotnet/tools:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
    DISPLAY: :99.0
  Verifying dependencies for moe@0.4.0
    Reading official package list

.
.
.
.

  [OK] Case 5: if the current line end with and in Python
  [OK] Case 5: if the current line end with : in Python
Error: execution of an external program failed: '/home/runner/work/moe/moe/tests/teditor '
     Error: Execution failed with exit code 1
        ... Command: /home/runner/.nimble/bin/nim c --noNimblePath -d:NimblePkgVersion=0.4.0 --path:/home/runner/.nimble/pkgs/ncurses-1.0.2 --path:/home/runner/.nimble/pkgs/unicodedb-0.10.0 --path:/home/runner/.nimble/pkgs/parsetoml-0.6.0 -r --path:. /home/runner/work/moe/moe/tests/teditor
X connection to :99 broken (explicit kill or server shutdown).
Error: Process completed with exit code 1.
```

--- a/.github/workflows/test.yaml
+++ b/.github/workflows/test.yaml
@@ -71,7 +71,7 @@ jobs:
     - uses: actions/checkout@v3.5.3
     - run: |
         sudo apt-get update -yqq
-        sudo apt-get install -y gcc libncursesw5-dev build-essential
+        sudo apt-get install -y gcc libncursesw5-dev build-essential xvfb x11-xserver-utils xsel xclip
     - name: Cache choosenim
       id: cache-choosenim
       uses: actions/cache@v3.3.1
@@ -99,7 +99,7 @@ jobs:
     - name: Run unit test
       run: |
         nimble refresh
-        nimble test --verbose -y
+        xvfb-run nimble test --verbose -y
 
     - name: Install moe
       run: nimble install --verbose -y

--- a/changelog.d/20230707_122750_GitHub_Actions_fix#1482.rst
+++ b/changelog.d/20230707_122750_GitHub_Actions_fix#1482.rst
@@ -1,0 +1,7 @@
+.. _#1749:  https://github.com/fox0430/moe/pull/1749
+
+Fixed
+.....
+
+- `#1749`_ Fix #1482
+

--- a/tests/tclipboard.nim
+++ b/tests/tclipboard.nim
@@ -23,14 +23,17 @@ import moepkg/[settings, unicodeext, clipboard]
 import moepkg/platform {.all.}
 import moepkg/independentutils {.all.}
 
+proc isXAvailable(): bool {.inline.} =
+  execCmdExNoOutput("xset q") == 0
+
 proc isXselAvailable(): bool {.inline.} =
-  execCmdExNoOutput("xset q") == 0 and execCmdExNoOutput("xsel --version") == 0
+  isXAvailable() and execCmdExNoOutput("xsel --version") == 0
 
 proc isXclipAvailable(): bool {.inline.} =
-  execCmdExNoOutput("xset q") == 0 and execCmdExNoOutput("xclip -version") == 0
+  isXAvailable() and execCmdExNoOutput("xclip -version") == 0
 
 if isXselAvailable():
-  suite "Editor: Send to clipboad (xsel)":
+  suite "Clipboard: Send to clipboad (xsel)":
     test "Send string to clipboard 1 (xsel)":
       const
         buffer = @[ru "Clipboard test"]
@@ -107,7 +110,7 @@ if isXselAvailable():
           check output[0 .. output.high - 2] == $buffer
 
 if isXclipAvailable():
-  suite "Send string to clipboard (xclip)":
+  suite "Clipboard: Send string to clipboard (xclip)":
     test "Send string to clipboard (xclip)":
       const
         buffer = @[ru "`````"]

